OnionUI/Onion
# Multiple changes to NTP (and time related services)

## Changes

Was having weird issues with time advancing 1 hour with ntpd, compiled ntpdate which seems to work ok.

- Now uses ntpdate instead of ntpd
- Fixes ntpdate on main (in runtime.sh) (Sets TZ in Main)
- Fixes ntpdate in telnet (in update_networking.sh)
- Adds file launchtelnet.sh to script dir
- Runs telnet (when used from tweaks) with some env vars commonly used throughout onion to fix telnet time/date
- Optimised ntpdate functions
- THERE IS A DELAY FOR TIME TO UPDATE. (AROUND 5-10 SECONDS)
- When integrating NTP into your SH environment (if for some reason it doesn't get the TZ from Main), use `export TZ=$(cat "$sysdir/config/T.Z")`
- Fixed an issue that caused hotspot not to work after making the above changes (???)

## Flow:

- User turns on NTP in Tweaks
- update_networking runs check_ntpstate which:
	- gets the tzid,
	- sets the relevent POSIX TZ,
	- sets the hwclock
	- creates a tracker file
- update_networking runs set_tzid which:
	- exports the TZ into the current shell (this sets the telnet/ftp/ssh/filebrowser time)
	- writes the TZ to a file located at $sysdir/config/T.Z
- runtime.sh checks for the tracker file, if it exists:
	- exports the TZ found in $sysdir/config/T.Z to set the MainUI TZ (which sets the clock and anything else started by MainUI)
	-  NTP Waits for shutdown or TZ to be changed.

## Test process PLEASE VALIDATE:

- Clean boot (FTP/Telnet/SSH/Filebrowser/NTP turned OFF, restart)
- Turn off Emulated time skip
- Open clock app, note clock says incorrect time
- Enter tweaks, turn on Telnet
- Open Telnet session, run printenv and notice your TZ is set
- Run `date`, note time incorrect
- Run `hwclock`, note time incorrect
- Close Telnet session
- Enter tweaks, turn on NTP, Exit tweaks
- WAIT 5-10 seconds
- Open clock app, note time is set
- Open telnet, run `printenv`, note TZ is set.
- Run `date`, note time is correct
- Run `hwclock`, note time is correct
- run: `touch test.test` and check the file time with ftp/stat
- Close telnet
- Open tweaks, change TZ to something else, close tweaks
- WAIT 5-10 secs
- Open clock app, note time is set 
Cycle.

## Issues:

-bftpd for some reason (even though it's started in the same shell as ntp) is 1 hour ahead. I think this is due to MainUI advancing time by 1 hour and i can't figure out the cause, anything that reads hwclock is correct. Anything that reads date weirdly gets advanced in my testing 
-shells dont get updated with a new TZ if they've already started with the old one, can't do anything about this unless each shell checks for a new TZ in a loop.
